mergify_engine/rules.py

```diff
@@ -133,7 +133,8 @@ def get_branch_rule(g_repo, branch, ref=github.GithubObject.NotSet):
     dict_merge(rule, rules.get("default", {}))
 
     for branch_re in rules.get("branches", []):
-        if re.match(branch_re, branch):
+        if (branch_re == branch or
```

Comment:
This looks like there might be a problem if a branch has the same name of the regexp.
Something like:

```python
if (branch_re[0] == "^" and re.match(branch_re, branch)) or (branch_re[0] != "^" and branch_re == branch)
```

would be more solid.

Reply:
good catch